sisl/grid.py

```diff
@@ -195,6 +218,76 @@ def interp(self, shape, method='linear', **kwargs):
 
         return grid
 
+    def smooth(self, method='gaussian', r=None, mode='wrap', **kwargs):
+        '''
+        Make a smoother grid by applying a filter.
+
+        Parameters
+        -----------
+        method: {'gaussian', 'uniform'}, optional
+            the type of filter to apply to smoothen the grid.
+        r: float or array-like of float, optional
+            the radius of the filter in Angstrom for each axis.
+            If the method is 'gaussian', this is the standard deviation!
+            
+            If a single float is provided, then the same distance will be used for all axes.
+            
+            The defaults are 0.6 Ang for a gaussian filter and 1.5 Ang for a uniform filter.
+        mode: {'wrap', 'mirror', 'constant', 'reflect', 'nearest'}, optional
+            determines how to compute the borders of the grid.
+            The default is wrap, which accounts for periodic conditions. 
+            
+            See https://docs.scipy.org/doc/scipy/reference/generated/scipy.ndimage.gaussian_filter.html#scipy.ndimage.gaussian_filter
+        '''
+        
+        if r is None:
+            r = 0.6 if method == 'gaussian' else 1.5
+        # Normalize the radius input to a list of radius
+        if isinstance(r, (int, float)):
+            r = [r]*3
+        
+        # Calculate the size of the kernel in pixels (in case the
+        # gaussian filter is used, this is the standard deviation)
+        size = (r/np.linalg.norm(self.dcell, axis=1)).astype(int)
+        
+        # Update the kwargs accordingly
+        if method == 'gaussian':
+            kwargs['sigma'] = size  
+        elif method == 'uniform':
+            kwargs['size'] = size
+        
+        # Apply the method
+        return self.apply(f'{method}_filter', mode=mode, **kwargs)
+  
+    def apply(self, method, *args, **kwargs):
+        '''
+        Applies a function of `scipy.ndimage` to the grid and returns a new grid.
```

Comment:
Correct documentation, you may apply *any* function, but a str converts to `scipy.ndimage.<>`

Reply:
Coming to think of it, I don't think we should allow `method` as `str`, it is rather obscure that it takes functions from `scipy.ndimage`, if anything it should be a string with the full module name + method such that one could also do:
`method="numpy.interpolation.interp` or whatever?

The use of `importlib` would be good here, it seems you are the expert in this ;)

Reply:
Or centralize a `str2method` method? in `sisl.utils` or something?

Reply:
> The use of importlib would be good here, it seems you are the expert in this ;)

Yes, sure, I'm a core developer of importlib :smile: 

> Or centralize a str2method method? in sisl.utils or something?

Sounds good to me!

Do you still think it's obscure if we have a keyword `module='scypy.ndimage'`?

Reply:
hmm... No I think that is bad practice since a function *could* use a keyword "module".
In any case it isn't hard for users to create this full string.

Reply:
Ok :+1: 

By the way, a function could also accept a keyword argument called "method" (actually, it is more likely than "module"), so should I rename it?

Reply:
hmm, yeah. Perhaps this is a good idea.

`method_` would be pretty good, what do you think?
Still explicit but hides it from the user since it will most likely be used non-keyword.

Reply:
Good? Should I call it `function_`, since it actually doesn't need to be a method?

Reply:
Agreed :)